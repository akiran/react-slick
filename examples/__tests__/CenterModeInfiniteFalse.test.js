import React from "react";
import { mount } from "enzyme";
import { html as beautify_html } from "js-beautify";
import { clickNext } from "../../__tests__/testUtils";
import CenterModeInfiniteFalse from "../CenterModeInfiniteFalse";

describe("CenterModeInfiniteFalse Tests", () => {
  test("Counting test", () => {
    const slider = mount(<CenterModeInfiniteFalse />);
    let totalSlides = slider.find(".slick-slide").length;
    let clonedSlides = slider.find(".slick-cloned").length;
    let activeSlides = slider.find(".slick-slide.slick-active").length;
    expect(totalSlides).toEqual(6);
    expect(clonedSlides).toEqual(0);
    expect(activeSlides).toEqual(3);
    expect(beautify_html(slider.html())).toMatchSnapshot();
  });
  test("Positioning test", () => {
    const slider = mount(<CenterModeInfiniteFalse />);
    let currentSlide = slider.find("div.slick-current");
    let activeSlides = slider.find("div.slick-active");
    expect(currentSlide.props()["data-index"]).toEqual(1);
    expect(activeSlides.map(slide => slide.props()["data-index"])).toEqual([
      0,
      1,
      2
    ]);
    expect(beautify_html(slider.html())).toMatchSnapshot();
  });
  test("Activity test", () => {
    const slider = mount(<CenterModeInfiniteFalse />);
    let currentSlide = slider.find("div.slick-current");
    let activeSlides = slider.find("div.slick-active");
    expect(currentSlide.props()["data-index"]).toEqual(1);
    expect(activeSlides.map(slide => slide.props()["data-index"])).toEqual([
      0,
      1,
      2
    ]);

    clickNext(slider);

    currentSlide = slider.find("div.slick-current");
    activeSlides = slider.find("div.slick-active");
    expect(currentSlide.props()["data-index"]).toEqual(2);
    expect(activeSlides.map(slide => slide.props()["data-index"])).toEqual([
      1,
      2,
      3
    ]);

    expect(beautify_html(slider.html())).toMatchSnapshot();
  });
});
