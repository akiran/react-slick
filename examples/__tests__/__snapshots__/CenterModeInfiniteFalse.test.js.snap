// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterModeInfiniteFalse Tests Activity test 1`] = `
"<div>
    <h2>Center Mode</h2>
    <div class=\\"slick-slider center slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\" style=\\"padding: 0px 60px;\\">
            <div class=\\"slick-track\\" style=\\"width: -240px; opacity: 1; transform: translate3d(40px, 0px, 0px); transition: transform 500ms ease;\\">
                <div data-index=\\"0\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"2\\" class=\\"slick-slide slick-active slick-center slick-current\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"3\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"4\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"5\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
    </div>
</div>"
`;

exports[`CenterModeInfiniteFalse Tests Counting test 1`] = `
"<div>
    <h2>Center Mode</h2>
    <div class=\\"slick-slider center slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\" style=\\"padding: 0px 60px;\\">
            <div class=\\"slick-track\\" style=\\"width: -240px; opacity: 1; transform: translate3d(0px, 0px, 0px);\\">
                <div data-index=\\"0\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" class=\\"slick-slide slick-active slick-center slick-current\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"2\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"3\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"4\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"5\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
    </div>
</div>"
`;

exports[`CenterModeInfiniteFalse Tests Positioning test 1`] = `
"<div>
    <h2>Center Mode</h2>
    <div class=\\"slick-slider center slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\" style=\\"padding: 0px 60px;\\">
            <div class=\\"slick-track\\" style=\\"width: -240px; opacity: 1; transform: translate3d(0px, 0px, 0px);\\">
                <div data-index=\\"0\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" class=\\"slick-slide slick-active slick-center slick-current\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"2\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"3\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"4\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"5\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: -40px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
    </div>
</div>"
`;
